Dockerfile

```diff
@@ -17,4 +17,13 @@ COPY . /app
 
 COPY --from=opbeans/opbeans-frontend:latest /app/build /app/frontend/build
 
+LABEL \
+    org.label-schema.schema-version="1.0" \
+    org.label-schema.vendor="Elastic" \
+    org.label-schema.name="opbeans-ruby" \
+    org.label-schema.version="3.8.0" \
```

Comment:
It seems that this line will hold the version, right? Maybe we can move it out of this hidden place and directly retrieve the version from the agent itself, as we already do when bumping the elastic-apm dependency.

Maybe we can use a docker ARG when building the image, getting the above `label-schema.version` from there.

Reply:
This is just the metadata that gets updated with `.ci/bump-version.sh` when a new release version of the agent has been released.

Reply:
One last thing: I saw that we are adding the Dockerfile to git on each bump. I'm ok with that, no blocker on my side, just adding if using ARG then the file would keep untouched, having a cleaner git history. On the other hand, it's automated :) Just thinking out loud!

Reply:
Let me explain the current release process and the reasons for bumping the version in the Dockerfile too.

The release for the opbeans and push to docker registry happens for two reasons:

- When merging to master. The docker image will be tagged as `latest`.
- When the agent gets released, therefore there will be a commit with the agent version into the master branch and a new release tag will be created in this repo with the format `v<agent-version`. 

Both releases (master and tag based) use the command `VERSION=(<agent-version>|master) make push`, therefore there is a chance that the `VERSION` could match something different than <agent-version> or latest. For instance when running the script locally. If so, the `ARG` will be set incorrectly.

This opinionated process will ensure no logic is in place in the opbeans repo but the agent pipeline will be the one that triggers the release tags in this repo. 

And even if running the `VERSION=wrong make push` will push into the dockerhub with the tag `wrong` but the metadata will still be pointing to the right agent version.

I hope I clarify my concerns with the `ARG` approach based on the current release process that's orchestrated from the agent itself.

Reply:
I see then. Thanks for the clear explanation :)

![image](https://media.giphy.com/media/4Z3DdOZRTcXPa/giphy.gif)